drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -734,6 +734,7 @@ static int esp32_wifi_status(const struct device *dev, struct wifi_iface_status
 	}
 
 	strncpy(status->ssid, data->status.ssid, WIFI_SSID_MAX_LEN);
+	status->ssid[WIFI_SSID_MAX_LEN] = '\0';
```

Comment:
I am confused, AFAICT `status->ssid` is 32 bytes. currently, this change gives

```bash
/home/brandon/fork/zephyr/drivers/wifi/esp32/src/esp_wifi_drv.c: In function 'esp32_wifi_status':
/home/brandon/fork/zephyr/drivers/wifi/esp32/src/esp_wifi_drv.c:737:21: warning: array subscript 32 is above array bounds of 'char[32]' [-Warray-bounds]
  737 |         status->ssid[WIFI_SSID_MAX_LEN] = '\0';
```

Reply:
Actually in my previous submission, it was [WIFI_SSID_MAX_LEN-1]. But it seems the auditors had issues with it, so I changed it

Reply:
> @jukkar @krish2718 should we change the API from `char ssid[WIFI_SSID_MAX_LEN];` to `char ssid[WIFI_SSID_MAX_LEN + 1];` as a proper solution instead?
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/wifi_mgmt.h#L432 https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/wifi_mgmt.h#L432

Yes, that is correct. If we want to store null at the end, there needs to be space for it in the buffer (so +1 is needed).

Reply:
I think it's good to change it to WIFI_SSID_MAX_LEN-1. Instead of changing the definition, because WIFI_SSID_MAX_LEN already indicates the maximum length. If you +1 to the maximum length, it will make the code extremely difficult to understand.

Reply:
The standard for SSID is 32 bytes long. The way it is now, it will only be 31 bytes due to the '\0' so it must be increased if we want null termination

Reply:
If it's okay, I'll reopen a PR tomorrow. It'll change the size of the ssid member (even though I’m not a big fan of this, but if the ssid is the standard 32-byte length, then the ssid member has to be 33 bytes).